app.js

```diff
@@ -1,35 +1,173 @@
+                   ['Am I Megan?', 'Y', 'Yep, that\'s my name!', 'Seriously?']];
+
+var userAns;
+var question;
+var correctAns;
+var correctResp;
+var wrongResp;
+
+for (var ask = 0; ask < ynQuestions.length; ask++) {
+  questionCount++;
+  question = ynQuestions[ask][0];
+  correctAns = ynQuestions[ask][1];
+  correctResp = ynQuestions[ask][2];
+  wrongResp = ynQuestions[ask][3];
+  do {
+    userAns = prompt(question);
+    if (typeof userAns === 'string') {
+      userAns = userAns.toUpperCase().trim();
+      userAns = userAns.charAt(0) === correctAns.charAt(0);
+    }
+    console.log(question, 'user correct:', userAns);
+  } while (typeof userAns != 'boolean')
+  if (userAns) {
+    alert(correctResp);
+    correctCount++;
   } else {
-    var tInput = input.toUpperCase().trim();
-    return tInput == 'YES' || tInput == 'Y';
+    alert(wrongResp);
   }
-};
+}
+console.log('correct y/n answers:', correctCount);
 
-/**
-Takes the question to be asked in a prompt, the answer for a response of 'yes', and the answer for a response of 'no'. Asks the question until the 'OK' button is pressed. Prints the user response as a boolean or null to the console. Responds to the user with the given answer in an alert. Returns nothing.
-**/
-var askQuestion = function(question, yesAns, noAns) {
-  var response;
-  do {
-    response = checkInput(prompt(question));
-    console.log(question, response);
-  } while (typeof response != 'boolean')
-  if (response) {
-    alert(yesAns);
+// GUESS THE NUMBER QUESTIONS
+
+// userAns defined above
+
+var favNum = Math.round((Math.random() * 100));
+console.log('my favorite number is:', favNum);
+questionCount++;
+
+var isNotDone = true;
+var message;
+
+for (var guess = 3; guess >= 0 && isNotDone; guess--) {
+
+  userAns = prompt('Guess my favorite number!');
+  console.log('user gueses:',userAns);
+  if (isNaN(userAns)) {
+    alert('That\'s not a number...')
   } else {
-    alert(noAns);
+    userAns = parseInt(userAns);
+    isNotDone = userAns != favNum;
+
+    if (isNotDone) {
+      message = 'No, that number is too ';
+      if (userAns > favNum) {
+        message += 'large.';
+      } else {
+        message += 'small.';
+      }
+      if (guess != 0) {
+        message += '\nOnly ' + guess + ' ';
+        if (guess != 1) {
+          message += 'guesses remain.';
+        } else {
+          message += 'guess remains.';
+        }
+      } else {
+        message += '\nToo bad, all out of guesses.\nIt was ' + favNum + '.';
+      }
+      alert(message);
+
+    } else {
+      alert('Yay! You guessed right!');
+      correctCount++;
+    }
+
+  }
+}
+
+console.log('current correct answers:', correctCount);
+
+// MULTIPLE ANSWER QUESTIONS
+
+// userAns defined above
+// message defined above
+var possibleAns = ['Canada', 'Mexico', 'France', 'England', 'Wales', 'Scotland'];
+questionCount++;
+isNotDone = true; // isNotDone defined above
+
+for (var guess = 6; guess >= 0 && isNotDone; guess--) {
```

Comment:
var guess is defined two times in app.js. because it's in a for loop but not a function, it's available on the global scope